lib/std/fs.zig

```diff
@@ -1084,14 +1089,16 @@ pub const Dir = struct {
             try os.openatZ(self.fd, sub_path_c, os_flags, flags.mode);
         errdefer os.close(fd);
 
-        if (!has_flock_open_flags and flags.lock != .None) {
-            // TODO: integrate async I/O
-            const lock_nonblocking = if (flags.lock_nonblocking) os.LOCK_NB else @as(i32, 0);
-            try os.flock(fd, switch (flags.lock) {
-                .None => unreachable,
-                .Shared => os.LOCK_SH | lock_nonblocking,
-                .Exclusive => os.LOCK_EX | lock_nonblocking,
-            });
+        if (builtin.target.os.tag != .wasi) {
+            if (!has_flock_open_flags and flags.lock != .None and builtin.target.os.tag != .wasi) {
```

Comment:
```suggestion
        if (!has_flock_open_flags and flags.lock != .None and builtin.target.os.tag != .wasi) {
```

I don't think the outer check is needed here.

Reply:
https://github.com/ziglang/zig/pull/9227#discussion_r688492678 same here